traitsui/qt4/text_editor.py

```diff
@@ -84,6 +84,13 @@ def init(self, parent):
             # updated.
             control.editingFinished.connect(self.update_object)
 
+        placeholder = self.factory.placeholder
+        try:
+            control.setPlaceholderText(placeholder)
+        except AttributeError:
+            # setPlaceholderText is introduced to QTextEdit since Qt 5.2
+            pass
+
```

Comment:
This is OK, but I have a slight preference for "look before you leap" here - AttributeErrors could come from a bug where the wrong control is being created, for example.  You want something like:
```
if wtype == QtGui.QLineEdit or QtCore.__version_info__ >= (5, 2):
    control.setPlaceholderText(self.factory.placeholder)
```
It's possibly good enough to just use `pyface.qt.is_qt5` instead of `QtCore.__version_info__ >= (5, 2)`.

Reply:
I did this very slightly differently hope that's okay:
```
if wtype is not QtGui.QTextEdit or QtCore.__version_info__ >= (5, 2):
```